addons/account_peppol/wizard/account_move_send.py

```diff
@@ -117,13 +117,11 @@ def _call_web_service(self, prepared_data_list):
             if not move.partner_id.peppol_eas or not move.partner_id.peppol_endpoint:
                 # should never happen but in case it does, we need to handle it
                 move.peppol_move_state = 'error'
-                move._message_log(body=_('The partner is missing Peppol EAS and/or Endpoint identifier.'))
-                continue
+                raise UserError(_('The partner is missing Peppol EAS and/or Endpoint identifier.'))
 
             if not move.partner_id.account_peppol_is_endpoint_valid:
                 move.peppol_move_state = 'error'
-                move._message_log(body=_('Please verify partner configuration in partner settings.'))
-                continue
+                raise UserError(_('Please verify partner configuration in partner settings.'))
```

Comment:
Suppose 3 invoices A, B and C, all of them send to Peppol. You get an error for A. Is it not an issue to skip the update of B and C?
I think we should update all invoices before raising. Is it possible Peppol return an error for A but not for the others?

Reply:
Do you mean we should update the `peppol_move_state` for all invoices with that partner instead of looping over the moves? :thinking: These moves will not be sent to Peppol.
You might not get an error instantly from Peppol, so I don't think we could do this based on what Peppol returns